src/poller.ts

```diff
@@ -0,0 +1,97 @@
+import { DIPDeposit } from "./common";
+import {
+  Connection,
+  AccountChangeCallback,
+  PublicKey,
+  AccountInfo as solanaAccountInfo,
+  Context,
+  clusterApiUrl,
+} from "@solana/web3.js";
+import { AccountInfo, AccountLayout, u64 } from "@solana/spl-token";
+
+export class Poller {
+  cluster: string;
+  callback: (deposit: DIPDeposit) => void;
+  splToken: string;
+  premiumAsset: string;
+  expiration: number;
+  strike: number;
+  type: string;
+
+  constructor(cluster: string, callback: (deposit: DIPDeposit) => void) {
+    this.cluster = cluster;
+    this.callback = callback;
+    this.splToken = "BTC";
+    this.premiumAsset = "USDC";
+    this.expiration = Date.now();
+    this.strike = 25000;
+    this.type = "call";
+  }
+
+  subscribe(address: string): void {
+    console.log("Listening at:", address);
+    let previous_amount = 0;
+    const connection: Connection = new Connection(clusterApiUrl("devnet"));
+    const callback: AccountChangeCallback = (
+      accountInfo: solanaAccountInfo<Buffer>,
+      _context: Context
+    ) => {
+      // @ts-ignore
+      let new_amount = parseTokenAccount(accountInfo.data).amount.toNumber();
+
+      const deposit_amount = new_amount - previous_amount;
+      const dip_deposit = {
+        splToken: this.splToken,
+        premiumAsset: this.premiumAsset,
+        expiration: this.expiration,
+        strike: this.strike,
+        type: this.type,
+        qty: deposit_amount,
+      };
+
+      // The amount could reduce because of exercise or withdraw.
```

Comment:
explain how you're treating expiry? why does withdrawal matter?

Reply:
We can either watch the token account for deposited spl tokens or option account. Might be better to watch the option account since that will not care about whether we exercise or withdraw

Reply:
If we watch the option account, then we need to differentiate between positions belonging to the MM or RM? 

Best design is one where the option tokens get minted to the option account then the router passes the tokens onto another account. If we track that account, then exercise will be obvious (tokens burned). If DIP tokens are ever withdrawn for some reason (maybe the protocol has too much risk and we need to sell some OTC) then it will adjust the hedging as necessary.

Reply:
Will address that when starting to branch to the market makers. The extra layer in the middle we have with the router should make this easy